src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx

```diff
@@ -0,0 +1,268 @@
+    await this.langSwitcherNode.refresh();
+    return Promise.resolve();
+  }
+
+  private async _handleSelection(e: CustomEvent): Promise<void> {
+    this._set("selectedLanguageData", e.detail);
+    console.log(e.detail);
+
+    const defaultLanguage = this._getDefaultLanguage();
+    if (e?.detail?.data === defaultLanguage || e?.detail === null) {
+      const templateAppData = await this._portalItem.fetchData();
+      if (
+        this._isWithinConfigurationExperience() &&
+        templateAppData?.values?.hasOwnProperty("draft")
+      ) {
+        Object.assign(this.configurationSettings, {
+          ...templateAppData?.values,
+          ...templateAppData?.values?.draft
+        });
+      } else {
+        Object.assign(this.configurationSettings, {
+          ...templateAppData?.values
+        });
+      }
+    }
+  }
+
+  private _getDefaultLanguage(): string {
+    // User profile - locale set in user profile
+    const userProfileLocale: string = this.base.portal?.get("user.culture");
+    // Browser - window.navigator.language
+    const browserLocale: string = window?.navigator?.language;
+    // ArcGIS JS API - locale currently set in JS api
+    const jsapiLocale: string = getLocale();
+    // Fallback locale - "en"
+    const fallbackLocale = "en";
+    return normalizeMessageBundleLocale(
+      userProfileLocale || browserLocale || jsapiLocale || fallbackLocale
+    ) as string;
+  }
+
+  private async _languageSwitcherCallback(
+    widgetProps: esriWidgetProps,
+    propertyName: string
+  ): Promise<void> {
+    widgetProps.propertyName = propertyName;
+    const languageSwitcher = await this._addLanguageSwitcher(widgetProps);
+    const handleKey = "languageSwitcher";
+
+    const handleExists = this.handles.has(handleKey);
+
+    when(
+      () => languageSwitcher,
+      () => {
+        when(
+          () => this.selectedLanguageData,
+          () => {
+            this._setLanguageSwitcherUI(this.base.config, this.configurationSettings);
+          },
+          { initial: true, once: true }
+        );
+      },
+      { initial: true, once: true }
+    );
+
+    if (!handleExists) {
+      this.handles.add(
+        watch(
+          () => this.selectedLanguageData,
+          () => this._setLanguageSwitcherUI(this.base.config, this.configurationSettings),
+          { initial: true }
+        ),
+        handleKey
+      );
+    }
+  }
+
+  private async _languageSwitcherConfigCallback(widgetProps: esriWidgetProps): Promise<void> {
+    const expand = widgetProps?.view?.ui?.find("esri-language-switcher") as Expand;
+    expand.expandIcon = this.configurationSettings.languageSwitcherConfig.icon;
+    this.configurationSettings.languageSwitcherConfig =
+      this.configurationSettings.languageSwitcherConfig;
+    await this._refresh();
+    this._setLanguageSwitcherUI(this.base.config, this.configurationSettings);
+  }
+
+  private _isWithinConfigurationExperience(): boolean {
```

Comment:
Looks like we have this in app base in the TCL. Might make sense to move isWIthinConfigurationExperience to a utils class?

Reply:
Yep makes sense - just moved it over to a separate util function